packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1423,6 +1424,9 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   /// {@endtemplate}
   bool get barrierDismissible;
 
+  /// The delegated transition provided to the previous route.
+  final DelegatedTransitionBuilder? delegatedTransition;
```

Comment:
I mentioned this above in the example, but maybe this should be a method that is supposed to be overridden instead of passed in.

Or better, instead of putting this into ModalRoute, create a mixin with a delegatedTransition. So MBSPageRoute would mix that in. Navigator.push would check for the mixin instead of checking for ModalRoute.